kubectl-openebs/cli/command/describe/volume_info.go

```diff
@@ -99,112 +98,109 @@ func RunVolumeInfo(cmd *cobra.Command, vols []string, ns string) error {
-	out := make([]string, len(cvrInfo.Items)+2)
-	out[0] = "Name|Pool Instance|Status"
-	out[1] = "----|-------------|------"
-	for i, cvr := range cvrInfo.Items {
-		out[i+2] = fmt.Sprintf("%s|%s|%s",
-			cvr.ObjectMeta.Name,
-			cvr.Labels[cstortypes.CStorPoolInstanceNameLabelKey],
-			cvr.Status.Phase,
-		)
+	for _, volName := range vols {
+		// Fetch all details of a volume is called to get the volume controller's
+		// info such as controller's IP, status, iqn, replica IPs etc.
+		//1. cStor volume info
+		volumeInfo, err := clientset.GetcStorVolume(volName)
+		if err != nil {
+			fmt.Printf("failed to get CStorVolume %s\n", volName)
+		}
+		//2. Persistent Volume info
+		pvInfo, err := clientset.GetPV(volName)
+		if err != nil {
+			fmt.Printf("failed to get PV %s\n", volName)
+		}
+		//3. cStor Volume Config
+		cvcInfo, err := clientset.GetCVC(volName)
+		if err != nil {
+			fmt.Printf("failed to get cStor Volume config for %s", volName)
+		}
+
+		//4. Get Node for Target Pod from the openebs-ns
+		nodeName, err := clientset.NodeForVolume(volName)
+		if err != nil {
+			fmt.Printf("failed to get CStorVolumeAttachments for %s\n", volName)
+		}
+
+		//5. cStor Volume Replicas
+		cvrInfo, err := clientset.GetCVR(volName)
+		if err != nil {
+			fmt.Printf("failed to get cStor Volume Replicas for %s\n", volName)
+		}
+		cSPCLabel := cstortypes.CStorPoolClusterLabelKey
+		volume := util.VolumeInfo{
+			AccessMode:              util.AccessModeToString(pvInfo.Spec.AccessModes),
+			Capacity:                volumeInfo.Status.Capacity.String(),
+			CSPC:                    cvcInfo.Labels[cSPCLabel],
+			CSIDriver:               pvInfo.Spec.CSI.Driver,
+			CSIVolumeAttachmentName: pvInfo.Spec.CSI.VolumeHandle,
+			Name:                    volumeInfo.Name,
+			Namespace:               volumeInfo.Namespace,
+			PVC:                     pvInfo.Spec.ClaimRef.Name,
+			ReplicaCount:            volumeInfo.Spec.ReplicationFactor,
+			VolumePhase:             pvInfo.Status.Phase,
+			StorageClass:            pvInfo.Spec.StorageClassName,
+			Version:                 util.CheckVersion(volumeInfo.VersionDetails),
+			Size:                    volumeInfo.Status.Capacity.String(),
+			Status:                  volumeInfo.Status.Phase,
+		}
+
+		// Print the output for the portal status info
+		tmpl, err := template.New("volume").Parse(volInfoTemplate)
+		if err != nil {
+			return errors.Wrap(err, "error displaying output for volume info")
+		}
+		err = tmpl.Execute(os.Stdout, volume)
+		if err != nil {
+			return errors.Wrap(err, "error displaying volume details")
+
+		}
+
+		portalInfo := util.PortalInfo{
+			IQN:            volumeInfo.Spec.Iqn,
+			VolumeName:     volumeInfo.Name,
+			Portal:         volumeInfo.Spec.TargetPortal,
+			TargetIP:       volumeInfo.Spec.TargetIP,
+			TargetNodeName: nodeName,
+		}
+
+		// Print the output for the portal status info
+		tmpl, err = template.New("PortalInfo").Parse(portalTemplate)
```

Comment:
- We have a util method that takes a generic template and does this job, that could be used here.-
Ref: https://github.com/openebs/openebsctl/pull/17/files#diff-b771e541bd0847ee044ef3d69f9875f032d4e2c427c89bac3f82b179b862bcb9